### PR TITLE
Tcs 115 ds stripe function updates

### PR DESCRIPTION
**Pull request by SeanW17**
Removes the need for the second variable in the .ds.stripe function.
Also defines .ds.numseg in the function rather than it being hardcoded.

### CODE REVIEW

File: code/common/subscriptions.q [hunk 1]
```diff
@@ -109,7 +109,11 @@ subscribe:{[tabs;instrs;setschema;replaylog;proc]
   // datastriping check on the tickerplant
   .ds.datastripe:@[proc`w;({@[value;`.ds.datastripe;0b]};`);`];
   .lg.o[`subscribe;"datastriping is turned ",$[.ds.datastripe;"on";"off"]];
- 
+  // define segmentlist from tickerplant definition
+  if[.ds.datastripe;
+	.ds.segmentlist:@[proc`w;({@[value;`.ds.segmentlist;0b]};`);`];
+	];
```
Comment (emansanti) [inline code/common/subscriptions.q:115]:
```suggestion
  ];
```

File: code/segmentedtickerplant/datastripe.q [hunk 2]
```diff
@@ -22,8 +22,11 @@ jsonchecks:{[]
      //Enable datastriping if all checks pass
      .ds.datastripe:min (0=count errors; 0=count wrongtables);
      $[.ds.datastripe;
-      .lg.o[`jsonchecks;"config checks complete and datastriping is on"];
-      .lg.e[`jsonchecks;"Datastriping is not enabled due to failed config checks"]
+	[.lg.o[`jsonchecks;"config checks complete and datastriping is on"];
```
Comment (emansanti) [inline code/segmentedtickerplant/datastripe.q:25]: Indentation in this addition seems quite messy and a bit inconsistent

Comment (SeanW17) [reply code/segmentedtickerplant/datastripe.q:25]: I changed it to use tabs but most of the code uses spaces i believe. I should probably just keep it consistent.

File: code/common/pubsub.q [hunk 3]
```diff
@@ -187,4 +193,4 @@ init:{[t]
     val:.ds.subreq input;
   ];
   skey=val
- };
```
Comment (emansanti) [inline code/common/pubsub.q:190]: Was this deleting a spare line at the end of the file?

File: code/common/pubsub.q [hunk 4]
```diff
@@ -174,9 +174,15 @@ init:{[t]
 
 // Initialise subscription request on startup
 .ds.subreq:(`u#`$())!`int$();
+
 // Striping function which stores the mappings for any symbols that it has already computed and 
 // for subsequent requests for that symbol, it looks them up
-.ds.stripe:{[input;skey]
+.ds.stripe:{[input]
+  // Define number of segments and match each segid to a section
+  .ds.numseg:`int$count .ds.segmentlist;
```
Comment (emansanti) [inline code/common/pubsub.q:182]: `numseg` doesn't seem to be used anywhere?

File: code/common/pubsub.q (hunk 3 above)
Comment (SeanW17) [reply code/common/pubsub.q:190]: I believe this was adding a new line to remove this warning in git diffs.

File: code/common/pubsub.q [hunk 5]
```diff
@@ -174,9 +174,15 @@ init:{[t]
...
+  .ds.segsect:.ds.segmentlist!til count .ds.segmentlist;
+  // Define segment based on sym split
+  skey:first .ds.segsect `$.proc.params[`segid];
```
Comment (jonathonmcmurray) [inline code/common/pubsub.q:185]: I could be mistaken, but I thought we got rid of the `-segid` command line param? Are we re-introducing it? Or has it always been there still?

File: code/common/pubsub.q [hunk 6]
```diff
@@ -174,9 +174,15 @@ init:{[t]
 
 // Initialise subscription request on startup
 .ds.subreq:(`u#`$())!`int$();
+
 // Striping function which stores the mappings for any symbols that it has already computed and 
 // for subsequent requests for that symbol, it looks them up
-.ds.stripe:{[input;skey]
+.ds.stripe:{[input]
+  // Define number of segments and match each segid to a section
```
Comment (jonathonmcmurray) [inline code/common/pubsub.q:181]: what is a "section" and how is it different to a "segment"?

File: code/segmentedtickerplant/datastripe.q (hunk 2 above)
Comment (jonathonmcmurray) [reply code/segmentedtickerplant/datastripe.q:25]: some lines here still using tabs, others spaces

File: code/common/pubsub.q (hunk 5 above)
Comment (SeanW17) [reply code/common/pubsub.q:185]: No, I believe we still use the segid, it's the .ds.datastripe variable that we got rid of in the command line. We may be getting rid of it in future if we sort out the extras to the json and such like but it's still in for now.

File: code/common/pubsub.q (hunk 6 above)
Comment (SeanW17) [reply code/common/pubsub.q:181]: I don't know why I worded it like this. I'll get that changed.
```suggestion
  // Define number of segments and match to each segid
```

File: code/common/subscriptions.q [hunk 7]
```diff
@@ -109,7 +109,11 @@ subscribe:{[tabs;instrs;setschema;replaylog;proc]
   // datastriping check on the tickerplant
   .ds.datastripe:@[proc`w;({@[value;`.ds.datastripe;0b]};`);`];
   .lg.o[`subscribe;"datastriping is turned ",$[.ds.datastripe;"on";"off"]];
- 
+  // define segmentlist from tickerplant definition
+  if[.ds.datastripe;
+	.ds.segmentlist:@[proc`w;({@[value;`.ds.segmentlist;0b]};`);`];
```
Comment (jonathonmcmurray) [inline code/common/subscriptions.q:114]: this line is tab, surrounding are all spaces

**Review by jonathonmcmurray — APPROVED** [review]

**Pull request merged**